helix-term/src/ui/prompt.rs

```diff
@@ -550,13 +550,17 @@ impl Component for Prompt {
                 if let Some(register) = self.history_register {
                     let register = cx.editor.registers.get_mut(register);
                     self.change_history(register.read(), CompletionDirection::Backward);
+                    self.recalculate_completion(cx.editor);
+                    self.exit_selection();
                     (self.callback_fn)(cx, &self.line, PromptEvent::Update);
```

Comment:
I'm unsure about this `self.exit_selection()`, but it was always called together with `recalculate_completion` so I decided to add it here as well.